psl/psl-core/src/validate/validation_pipeline/validations/relation_fields.rs

```diff
@@ -140,39 +142,79 @@ pub(super) fn ignored_related_model(field: RelationFieldWalker<'_>, ctx: &mut Co
 pub(super) fn referential_actions(field: RelationFieldWalker<'_>, ctx: &mut Context<'_>) {
     let connector = ctx.connector;
     let relation_mode = ctx.relation_mode;
-    let msg = |action: ReferentialAction| {
-        let allowed_values = connector
-            .referential_actions(&relation_mode)
-            .iter()
-            .map(|f| format!("`{}`", f.as_str()))
-            .join(", ");
+
+    fn fmt_allowed_actions(allowed_actions: BitFlags<ReferentialAction>) -> String {
+        allowed_actions.iter().map(|f| format!("`{}`", f.as_str())).join(", ")
+    }
+
+    // validation template for relationMode = "foreignKeys"
+    let msg_foreign_keys = |action: ReferentialAction| {
+        let allowed_actions = connector.referential_actions(&relation_mode);
 
         format!(
             "Invalid referential action: `{}`. Allowed values: ({})",
             action.as_str(),
-            allowed_values,
+            fmt_allowed_actions(allowed_actions),
+        )
+    };
+
+    // validation template for relationMode = "prisma"
+    let msg_prisma = |action: ReferentialAction| {
+        let allowed_actions = connector.emulated_referential_actions(&relation_mode);
+
+        let additional_info = match action {
+            ReferentialAction::NoAction => {
+                if ctx
+                    .connector
+                    .supports_referential_action(&relation_mode, ReferentialAction::Restrict)
```

Comment:
This check looks like it relies on out-of-band information: `supports_referential_action(Restrict)` and `emulated_referential_actions()` should be in sync for the message to make sense. If that's correct, we should document that with a comment.

Reply:
`emulated_referential_actions()` and `supports_referential_action(Restrict)` are indeed in sync, as
- this is the definition of `supports_referential_action`:

```rust
    fn supports_referential_action(&self, relation_mode: &RelationMode, action: ReferentialAction) -> bool {
        match relation_mode {
            RelationMode::ForeignKeys => self.referential_actions().contains(action),
            RelationMode::Prisma => self.emulated_referential_actions(relation_mode).contains(action),
        }
    }
```

- `msg_prisma` is documented as the validation template for `relationMode = "prisma"`

Do you have a specific comment in mind that would be nice to add there?

Reply:
I misunderstood that, I was taking `supports_referential_action()` as "the connector in general supports the given referential action", but it has that conditional. Since this is equivalent, could we replace this check with `allowed_actions.contains(Restrict)`?